File: screenshot-simple.js
// Ejemplo simple para capturas de pantalla
const APP_CONFIG = {
  name: "Mi Aplicación",
  version: "1.0.0",
  debug: true, // Habilitar logs de debug
  timeout: 5000, // Timeout en milisegundos
  maxRetries: 3, // Número máximo de reintentos
};

/**
 * Clase principal para manejar operaciones de datos
 * @class DataManager
 */
class DataManager {
  constructor(name) {
    this.name = name;
    this.data = []; // Array para almacenar datos
    this.isProcessing = false; // Flag de estado
    this.stats = {
      total: 0,
      processed: 0,
      errors: 0,
    };
  }

  /**
   * Procesa un array de datos y retorna estadísticas
   * @param {Array<string>} dataArray - Array de strings a procesar
   * @returns {Object} Objeto con estadísticas del procesamiento
   */
  async processData(dataArray) {
    console.log("Iniciando procesamiento de datos...");

    // Validar entrada
    if (!Array.isArray(dataArray)) {
      throw new Error("El parámetro debe ser un array");
    }

    this.isProcessing = true;
    this.stats.total = dataArray.length;
    this.stats.processed = 0;
    this.stats.errors = 0;

    // Procesar cada elemento
    for (let i = 0; i < dataArray.length; i++) {
      const item = dataArray[i];

      try {
        // Validar el elemento
        if (this.validateItem(item)) {
          this.data.push(item); // Agregar a la lista
          this.stats.processed++;
        } else {
          this.stats.errors++;
          console.warn(`Elemento inválido en posición ${i}: ${item}`);
        }
      } catch (error) {
        this.stats.errors++;
        console.error(`Error procesando elemento ${i}:`, error);
      }
    }

    this.isProcessing = false;
    return this.getStats();
  }

  /**
   * Valida un elemento individual
   * @param {string} item - String a validar
   * @returns {boolean} True si es válido
   */
  validateItem(item) {
    // Verificar que no esté vacío
    if (!item || typeof item !== "string") {
      return false;
    }

    // Verificar longitud mínima
    if (item.trim().length < 2) {
      return false;
    }

    // Verificar que no contenga caracteres peligrosos
    const dangerousChars = ["<", ">", "&", '"', "'", "`"];
    for (const char of dangerousChars) {
      if (item.includes(char)) {
        return false;
      }
    }

    return true;
  }

  /**
   * Obtiene estadísticas del procesamiento
   * @returns {Object} Estadísticas actuales
   */
  getStats() {
    return {
      ...this.stats,
      successRate:
        this.stats.total > 0 ? this.stats.processed / this.stats.total : 0,
    };
  }
}

/**
 * Función principal del programa
 */
async function main() {
  try {
    // Crear instancia del manager
    const dataManager = new DataManager("Mi Data Manager");

    // Datos de ejemplo para procesar
    const sampleData = [
      "elemento1", // Primer elemento
      "elemento2", // Segundo elemento
      "", // Elemento vacío (debería fallar)
      "ab", // Elemento válido
      "elemento con <script>", // Elemento con caracteres especiales
      "elemento5", // Último elemento
    ];

    // Procesar los datos
    const results = await dataManager.processData(sampleData);

    // Mostrar resultados
    console.log("=== RESULTADOS DEL PROCESAMIENTO ===");
    console.log(`Total de elementos: ${results.total}`);
    console.log(`Procesados exitosamente: ${results.processed}`);
    console.log(`Errores encontrados: ${results.errors}`);
    console.log(`Tasa de éxito: ${(results.successRate * 100).toFixed(2)}%`);
  } catch (error) {
    console.error("Error en la función principal:", error);
    process.exit(1);
  }
}

// Ejecutar solo si es el archivo principal
if (require.main === module) {
  main();
}

module.exports = { DataManager };

File: screenshot-simple.py
# Ejemplo simple para capturas de pantalla
import os
import sys

# Configuración global
CONFIG = {
    'debug': True,  # Habilitar modo debug
    'timeout': 30,  # Timeout en segundos
    'retries': 3    # Número de reintentos
}


class DataProcessor:
    """
    Clase para procesar datos de manera eficiente
    Esta clase maneja la lógica de negocio principal
    """

    def __init__(self, name: str):
        self.name = name
        self.data = []  # Lista para almacenar datos
        self.processed = False  # Flag de procesamiento

    def process_data(self, data: list) -> dict:
        """
        Procesa una lista de datos y retorna estadísticas

        Args:
            data: Lista de strings a procesar

        Returns:
            Diccionario con estadísticas del procesamiento
        """
        # Inicializar contadores
        total_items = len(data)
        processed_items = 0
        errors = 0

        # Procesar cada elemento
        for item in data:
            try:
                # Validar el elemento
                if self._validate_item(item):
                    processed_items += 1
                    self.data.append(item)  # Agregar a la lista
                else:
                    errors += 1  # Incrementar contador de errores
            except Exception as e:
                # Log del error (en producción usar logger)
                print(f"Error procesando {item}: {e}")
                errors += 1

        # Marcar como procesado
        self.processed = True

        # Retornar estadísticas
        return {
            'total': total_items,
            'processed': processed_items,
            'errors': errors,
            'success_rate': processed_items / total_items if total_items > 0 else 0
        }

    def _validate_item(self, item: str) -> bool:
        """
        Valida un elemento individual

        Args:
            item: String a validar

        Returns:
            True si es válido, False en caso contrario
        """
        # Verificar que no esté vacío
        if not item or not item.strip():
            return False

        # Verificar longitud mínima
        if len(item) < 3:
            return False

        # Verificar que no contenga caracteres especiales
        special_chars = ['<', '>', '&', '"', "'"]
        for char in special_chars:
            if char in item:
                return False

        return True


def main():
    """
    Función principal del programa
    """
    # Crear instancia del procesador
    processor = DataProcessor("Mi Procesador")

    # Datos de ejemplo para procesar
    sample_data = [
        "elemento1",  # Primer elemento
        "elemento2",  # Segundo elemento
        "",           # Elemento vacío (debería fallar)
        "abc",        # Elemento válido
        "elemento con <script>",  # Elemento con caracteres especiales
        "elemento5"   # Último elemento
    ]

    # Procesar los datos
    print("Iniciando procesamiento...")
    results = processor.process_data(sample_data)

    # Mostrar resultados
    print(f"Total de elementos: {results['total']}")
    print(f"Procesados exitosamente: {results['processed']}")
    print(f"Errores encontrados: {results['errors']}")
    print(f"Tasa de éxito: {results['success_rate']:.2%}")


if __name__ == "__main__":
    main()
